doc/pages/source/documentation/index.md

```diff
@@ -268,6 +270,38 @@ def self.trigger_task(async, action, *args, &block)
 end
 ```
 
+#### Scheduling
+
+Scheduling an action means setting it up to be triggered at set time in future.
+Any action can be scheduled by calling:
+
+```ruby
+world_instance.schedule(AnAction, start_at, start_before, *args)
+```
+
+This snippet of code would schedule `AnAction` with arguments `args` to be executed
+in the time interval between `start_at` and `start_before`. Setting `start_before` to `nil`
+would schedule this action without the upper time limit.
+
+When an action is scheduled, minimal execution plan is created for it and a schedule record
```

Comment:
What about starting describing first how it behaves from the users perspective:
"When an action is scheduled, execution plan object is created (state = scheduled), but it doesn't run the plan phase yet: the planning happens when the `start_at` time comes."
Followed by what happens when the scheduler is down. Also, I would prefer marking the plan as stopped/error, instead of deleting it:
"When the planning doesn't happened before the `start_before` time, the execution plan is marked as failed (state = stopped, result = error)"

Then we could talk about the serialization and we would talk about the Scheduler itself just at the end, as this is more an implementation detail than something the user is really interested in.